src/Uno.UI/UI/Xaml/Controls/Layouter/Layouter.cs

```diff
@@ -443,6 +443,10 @@ internal void ArrangeChild(View view, Rect frame, bool raiseLayoutUpdated)
 				return;
 			}
 			frame = ApplyMarginAndAlignments(view, frame);
+			if (view is UIElement elt)
+			{
+				elt.LayoutSlotWithMarginsAndAlignments = frame;
```

Comment:
Why ?

Reply:
I use the `LayoutSlotWithMarginsAndAlignments` in the `TransformToVisual` to get the location of the element in the parent coordinates system. The issue is that the `LayoutSlot` from UWP does not contains the margins nor alignments applied by the parent element. I tried to reverse-compute it, but it's implies to down cast element (for instance for the margins I had to convert down cast to `FrameworkElement`, the scroll to `ScrollViewer` etc.), it's less performant and much more error prompt ... I found that the easiest / safest solution was to store the "native layout slot" (a.k.a. `LayoutSlotWithMarginsAndAlignments`) in the arrange.